src/jenkins_job_insight/storage.py

```
@@ -2894,3 +2932,199 @@ async def bulk_set_metadata(items: list[dict]) -> dict:
+
+        # Totals
+        totals_query = (
+            "SELECT "
+            "COALESCE(SUM(input_tokens), 0) as total_input_tokens, "
+            "COALESCE(SUM(output_tokens), 0) as total_output_tokens, "
+            "COALESCE(SUM(cache_read_tokens), 0) as total_cache_read_tokens, "
+            "COALESCE(SUM(cache_write_tokens), 0) as total_cache_write_tokens, "
+            "COALESCE(SUM(cost_usd), 0) as total_cost_usd, "
+            "COUNT(*) as total_calls, "
+            "COALESCE(SUM(duration_ms), 0) as total_duration_ms "
+            f"FROM ai_token_usage{where_clause}"
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)